xtable-service/src/main/java/org/apache/xtable/service/ConversionService.java

```diff
@@ -0,0 +1,156 @@
+   * the conversion controller. After conversion, it retrieves schema and metadata paths for each
+   * target table.
+   *
+   * @param convertTableRequest the conversion request containing source table details and target
+   *     formats
+   * @return a response containing details of converted target tables
+   */
+  public ConvertTableResponse convertTable(ConvertTableRequest convertTableRequest) {
+    SourceTable sourceTable =
+        SourceTable.builder()
+            .name(convertTableRequest.getSourceTableName())
+            .basePath(convertTableRequest.getSourceTablePath())
+            .formatName(convertTableRequest.getSourceFormat())
+            .build();
+
+    List<TargetTable> targetTables = new ArrayList<>();
+    for (String targetFormat : convertTableRequest.getTargetFormats()) {
+      TargetTable targetTable =
+          TargetTable.builder()
+              .name(convertTableRequest.getSourceTableName())
+              .basePath(convertTableRequest.getSourceTablePath())
+              .formatName(targetFormat)
+              .build();
+      targetTables.add(targetTable);
+    }
+
+    ConversionConfig conversionConfig =
+        ConversionConfig.builder().sourceTable(sourceTable).targetTables(targetTables).build();
+
+    ConversionSourceProvider<?> conversionSourceProvider =
+        conversionServiceUtil.getConversionSourceProvider(
+            convertTableRequest.getSourceFormat(), sparkHolder.jsc().hadoopConfiguration());
+
+    conversionController.sync(conversionConfig, conversionSourceProvider);
+
+    List<ConvertedTable> restTargetTables = new ArrayList<>();
+    for (String targetFormat : convertTableRequest.getTargetFormats()) {
+      if (targetFormat.equals(ICEBERG)) {
+        Pair<String, String> responseFields =
+            conversionServiceUtil.getIcebergSchemaAndMetadataPath(
+                convertTableRequest.getSourceTablePath(), sparkHolder.jsc().hadoopConfiguration());
+        ConvertedTable icebergTable =
+            new ConvertedTable(ICEBERG, responseFields.getLeft(), responseFields.getRight());
+        restTargetTables.add(icebergTable);
+      } else if (targetFormat.equals(HUDI)) {
+        Pair<String, String> responseFields =
+            conversionServiceUtil.getHudiSchemaAndMetadataPath(
+                convertTableRequest.getSourceTablePath(), sparkHolder.jsc().hadoopConfiguration());
+        ConvertedTable hudiTable =
+            new ConvertedTable(HUDI, responseFields.getLeft(), responseFields.getRight());
+        restTargetTables.add(hudiTable);
+      } else if (targetFormat.equals(DELTA)) {
+        Pair<String, String> responseFields =
+            conversionServiceUtil.getDeltaSchemaAndMetadataPath(
+                convertTableRequest.getSourceTablePath(), sparkHolder.spark());
+        ConvertedTable deltaTable =
+            new ConvertedTable(DELTA, responseFields.getLeft(), responseFields.getRight());
+        restTargetTables.add(deltaTable);
+      }
```

Comment:
See if we can use the getCurrentTable() method to avoid this code block. 
https://github.com/apache/incubator-xtable/blob/main/xtable-core/src/main/java/org/apache/xtable/conversion/ConversionController.java#L239

Reply:
will try taking a look at this.